### PR TITLE
First draft of ResourceElement - rewrite rendering layer

### PR DESCRIPTION
**Pull request by grahamegrieve**

### CODE REVIEW

File: org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+    assertEquals(children.get(2), obs.child("value"));
+    assertEquals(children.get(2), obs.child("value[x]"));
+    checkObsValue(children.get(2));
+    
+    assertEquals(children.get(0), obs.child("contained"));
+    checkContained(children.get(0));
+  }
+
+  private void checkContained(ResourceElement cont) {
```
Comment (dotasek) [inline org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java:142]: Should we check the reference value?

File: org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  }
+
+  private void checkObservation(ResourceElement obs) {
+    Assertions.assertTrue(obs.fhirType().equals("Observation"));
+    Assertions.assertEquals("resource", obs.name());
+    Assertions.assertEquals("Bundle.entry[0].resource", obs.path());
+    Assertions.assertEquals(ElementKind.BundleEntry, obs.kind());
+    
+    List<ResourceElement> children = obs.children();
+    assertEquals(3, children.size());
```
Comment (dotasek) [inline org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java]: Something doesn't seem quote right here. Are `id` and `text` not children?

File: org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    Assertions.assertEquals("Provenance", cont.fhirType());
+    Assertions.assertEquals("contained", cont.name());
+    Assertions.assertEquals("Bundle.entry[0].resource.contained[0]", cont.path());
+    Assertions.assertFalse(cont.isPrimitive());
+    Assertions.assertFalse(cont.hasPrimitiveValue());
+    Assertions.assertTrue(cont.hasChildren());
+    Assertions.assertEquals(ElementKind.ContainedResource, cont.kind());
+  }
+
+  private void checkObsValue(ResourceElement obsValue) {
```
Comment (dotasek) [inline org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java:152]: Should we check the value here?

**Review by dotasek — CHANGES_REQUESTED** [review]
There are some things about the tests that don't add up, or seem to be missing. See comments.

File: org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java (hunk 2 above)
Comment (grahamegrieve) [reply org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java]: I added those properties to the test case - will update the source for the test

File: org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java (hunk 3 above)
Comment (grahamegrieve) [reply org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java:152]: which value?

File: org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java (hunk 1 above)
Comment (grahamegrieve) [reply org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java:142]: I suppose we should

File: org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java (hunk 3 above)
Comment (dotasek) [reply org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java:152]: After the merge, but I did not want to leave this hanging. Should we check the id value?

```xml
<id value="obs1"/>
```

**Pull request merged**